design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -0,0 +1,466 @@
+      name: passtnet
+      queueCount: 1
+  ...
+```
+
+#### Pros
+
+- Manual configuration is not required
+- Provides backward compatibility
+
+#### Cons
+
+- Adds API
+- Exposes an implementation detail to VM owners which is probably not useful for them.
+- This field is not relevant in case the VM is using only secondary networks
+
+### Alternative 3 - Use annotations instead of K8s API
+
+A new annotation will be added to the VirtualMachineInstance object.
+It could represent either alternative 1 or 2 in annotation form.
+
+#### Pros
+
+- Manual configuration is not required
+- Provides backward compatibility
+
+#### Cons
+
+- Since information is going to be passed between KubeVirt's components it is not ideal to use semiformal API. 
+- Annotations usually flow from VirtualMachineInstance to Pod and not vice versa.
+
+### Alternative 4 - Use the existing Downward API mechanism
+
+Currently, the VMI controller sets the `kubevirt.io/network-info` annotation on the virt-launcher pod in two cases:
+
+1. The VMI has an SR-IOV NIC.
+2. The VMI uses a network binding plugin
+
+The annotation's value is derived fom Multus' network-status annotation and is injected into the virt-launcher pod
+using K8s' [Downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/) mechanism.
+
+A new field will be added to the `NetworkInfo` struct:
+
+```go
+type NetworkInfo struct {
+Interfaces []Interface `json:"interfaces,omitempty"`
+PrimaryPodInterfaceName string `json:primaryPodInterfaceName,omitempty`
+}
+```
+
+A new mechanism should be added between virt-handler and virt-launcher to share this information, because:
+
+1. The primary pod interface name is also required by virt-handler
+2. The virt-handler does not have access to pod objects
+
+#### Pros
+
+#### Cons
+
+- Requires a new mechanism to have the information in virt-handler.
+- All virt-launcher pods will have to use the downward API mechanism instead of just the subset using it today
+
+## Specifying the primary pod interface name
+
+### Adding a cluster-wide preferred pod NIC name
+
+A new field will be added in KubeVirt's cluster-wide network configuration, which will specify the preferred pod
+interface name that should be bound to the VM interface destined for pod network:
+
+```go
+type NetworkConfiguration struct {
+...
+PreferredPodPrimaryInterfaceName string `json:"preferredPodPrimaryInterfaceName,omitempty"`
+...
+}
+```
+
+#### API Example
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    network:
+      preferredPodPrimaryInterfaceName: special-nic
+```
+
+virt-handler and virt-launcher will be adjusted to prefer this value when trying to bind the primary network interface.
+They will both fallback to `eth0` in case they cannot find the preferred interface name.
+
+#### Pros
+
+- Does not require Multus to work
+
+#### Cons
+
+- Requires manual configuration
```

Comment:
It being manual or not is dependent on implementation (e.g. an operator can do this).

I would replace this with: The Cluster Admin or an Operator needs to know which interface name to use in advance. I.e. understand what network solution the cluster has and the admin interested in using.

Reply:
Done.